controller/workitem_blackbox_test.go

```diff
@@ -969,6 +969,139 @@ func (s *WorkItem2Suite) TestWI2UpdateSetReadOnlyFields() {
+					"bar": {
+						Label: "Type2 bar",
+						Type:  &workitem.SimpleType{Kind: workitem.KindInteger},
+					},
+					"fooBar": {
+						Label: "Type2 fooBar",
+						Type: workitem.EnumType{
+							BaseType:   workitem.SimpleType{Kind: workitem.KindString},
+							SimpleType: workitem.SimpleType{Kind: workitem.KindEnum},
+							Values:     []interface{}{"alpha", "beta", "gamma"},
+						},
+					},
+					"integer-or-float-list": {
+						Label: "Type2 integer-or-float-list",
+						Type: workitem.ListType{
+							SimpleType:    workitem.SimpleType{Kind: workitem.KindList},
+							ComponentType: workitem.SimpleType{Kind: workitem.KindFloat},
+						},
+					},
+				}
+			}
+			return nil
+		}),
+		tf.WorkItems(1, func(fxt *tf.TestFixture, idx int) error {
+			fxt.WorkItems[idx].Type = fxt.WorkItemTypes[0].ID
+			fxt.WorkItems[idx].Fields["integer-or-float-list"] = []int{101}
+			fxt.WorkItems[idx].Fields["fooo"] = 2.5
+			fxt.WorkItems[idx].Fields["fooBar"] = "open"
+			fxt.WorkItems[idx].Fields["bar"] = "hello"
+			fxt.WorkItems[idx].Fields["reporter"] = fxt.Identities[0].ID.String()
+			fxt.WorkItems[idx].Fields["assigned-to"] = []string{fxt.Identities[0].ID.String(), fxt.Identities[1].ID.String()}
+			fxt.WorkItems[idx].Fields[workitem.SystemDescription] = rendering.NewMarkupContentFromLegacy("description1")
+			return nil
+		}),
+	)
+	// when
+	u := minimumRequiredUpdatePayload()
+	u.Data.Attributes[workitem.SystemVersion] = fxt.WorkItems[0].Version
+	u.Data.ID = &fxt.WorkItems[0].ID
+	u.Data.Relationships = &app.WorkItemRelationships{
+		BaseType: newRelationBaseType(fxt.WorkItemTypes[1].ID),
+	}
+	svc := testsupport.ServiceAsUser("TypeChangeService", *fxt.Identities[0])
+	s.T().Run("ok", func(t *testing.T) {
+		_, newWI := test.UpdateWorkitemOK(t, svc.Context, svc, s.workitemCtrl, fxt.WorkItems[0].ID, &u)
+
+		assert.Equal(t, fxt.WorkItemTypes[1].ID, newWI.Data.Relationships.BaseType.Data.ID)
+		newDescription := newWI.Data.Attributes[workitem.SystemDescription]
+		assert.NotNil(t, newDescription)
+		// Type of old and new field is same
+		assert.NotContains(t, newDescription, fxt.WorkItemTypes[0].Fields["fooo"].Label)
+		assert.Contains(t, newDescription, fxt.WorkItemTypes[0].Fields["bar"].Label)
+		assert.Contains(t, newDescription, fxt.WorkItemTypes[0].Fields["fooBar"].Label)
+		assert.Equal(t, "alpha", newWI.Data.Attributes["fooBar"]) // First value of enum for field foobar
+		compareWithGoldenAgnostic(t, filepath.Join(s.testDir, "update", "workitem_type.res.payload.golden.json"), newWI)
+	})
+
+	s.T().Run("disallow update of field along with type", func(t *testing.T) {
+		u.Data.Attributes[workitem.SystemTitle] = "xyz"
+		// TODO (ibrahim) - Check type of error once error 422 has been added.
+		//https://github.com/fabric8-services/fabric8-wit/pull/2202#discussion_r210184092
+		test.UpdateWorkitemConflict(t, svc.Context, svc, s.workitemCtrl, fxt.WorkItems[0].ID, &u)
+	})
+
+	s.T().Run("unauthorized", func(t *testing.T) {
```

Comment:
[SUGGESTION] There should be a test to capture this scenario -> Space Admin which is not workitem creator is successfully able to change WIT of workitem.

Reply:
How do I make an identity a space owner? Can you point me to a similar test?